examples/experimental/dagster-dlift/dagster_dlift/translator.py

```diff
@@ -44,15 +54,35 @@ def environment_data(self) -> DbtCloudProjectEnvironmentData:
 
     def get_model_spec(self, data: DbtCloudContentData) -> AssetSpec:
         # This is obviously a placeholder implementation
-        return AssetSpec(key=data.properties["uniqueId"])
+        return AssetSpec(key=clean_asset_name(data.properties["uniqueId"]))
 
     def get_source_spec(self, data: DbtCloudContentData) -> AssetSpec:
         # This is obviously a placeholder implementation
-        return AssetSpec(key=data.properties["uniqueId"])
+        return AssetSpec(key=clean_asset_name(data.properties["uniqueId"]))
 
     def get_test_spec(self, data: DbtCloudContentData) -> AssetCheckSpec:
-        # This is obviously a placeholder implementation
-        return AssetCheckSpec(name=data.properties["uniqueId"], asset=data.properties["uniqueId"])
+        associated_data_per_parent = []
+        for parent in data.properties["parents"]:
+            if parent["resourceType"] == "model":
+                associated_data_per_parent.append(
+                    self.environment_data.models_by_unique_id[parent["uniqueId"]]
+                )
+            elif parent["resourceType"] == "source":
+                associated_data_per_parent.append(
+                    self.environment_data.sources_by_unique_id[parent["uniqueId"]]
+                )
+            else:
+                # Macros, etc. which we don't create assets for.
+                continue
+        # Haven't actually figured out if it's possible for there to be more than one "asset-creatable" entity per test
+        parent_specs = cast(
+            Sequence[AssetSpec], [self.get_spec(data) for data in associated_data_per_parent]
+        )
+        return AssetCheckSpec(
+            name=clean_asset_name(data.properties["uniqueId"]),
+            asset=parent_specs[0].key,
```

Comment:
is this arbitrary, or is there meaning to the parent order in the gql response? seems fine, just might be worth noting if this is a framework feature on their end or just something we do to satisfy "check needs a single linked asset"

Reply:
something we do to satisfy "check needs a single linked asset." See the above comment, but I'm not actually sure if it's possible for there to be more than one parent model/source for a given test. It seems like not, it's just a quirk of the data model.